Renault WakeUp

premultiply opened the pull request.

premultiply committed:
wip

premultiply commented:
@andig Macht das so vom Code halbwegs Sinn?
Ich bin da in diesem Vehicle-API-Sachen überhaupt nicht bewandert...

premultiply commented:
Wohl nicht.
In den Body muss wohl sowas wie `{"data":{"type":"ChargingStart","attributes":{"action":"start"}}}` wenn man https://github.com/snaptec/openWB/blob/979e6615945ed0cb93a6de6db1e1c8bd46a39f13/modules/soc_myrenault/zoenwake.py glauben schenken mag.

premultiply committed:
wip

andig committed:
Add POST support

andig commented:
Ich hab POST hinzugefügt, aber völlig ungetestet. Beim POST fehlen jetzt die Header- müssten wir schauen ob es ohne geht oder welche der da noch braucht.

chrissooo commented:
Hat das hier einen Stand wo ich testen könnte?

chrissooo commented:
Guten Morgen,

ich hatte es wieder. Das Auto war im Tiefschlaf (wie gesagt, wenn viele Stunden am Kabel und kein Ladevorgang, geht kein CP INterupt mehr) und das Aufwecken hat per CP Interupt nicht mehr funktioniert. Per App "Ladung fortsetzten" ging dann.

Jetzt die Frage. Kann ich in dem Zusammenhand irgendwie unterstützen / testen? Sagts bescheid :)

andig commented:
@chrissooo dieser PR muss getestet werden: läuft er fehlerfrei? Bewirkt das Wakeup das Erwünschte?

chrissooo commented:
Hi @andig, habe mir eine Version gebaut und werde nun die Tage testen. Habe dazu den Loadpoint WakeUp herausgenommen damit das kurzfristig ausschließlich darüber läuft. Ich berichte dann.

chrissooo commented:
Leider scheint es nicht richtig zu funktionieren. EInmal hat es geklappt. Dann kam aber auch folgender fehler kurz zuvor:
[lp-1  ] ERROR 2022/03/22 15:00:10 wake-up vehicle: Post "https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/00fcfbc7-633d-470e-84f3-98aac9b5e3ab/kamereon/kca/car-adapter/v1/cars/UU1DBG001MU027029/actions/charging-start?country=DE": net/http: request canceled (Client.Timeout exceeded while awaiting headers)

Jetzt kommt nur noch [lp-1  ] ERROR 2022/03/22 16:05:26 wake-up vehicle: unexpected status: 403

chrissooo commented:
Hier nochmal der komplette Trace:

`[renault] TRACE 2022/03/22 19:33:19 POST https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/00fcfbc7-633d-470e-84f3-98aac9b5e3ab/kamereon/kca/car-adapter/v1/cars/***/actions/charging-start?country=DE
[renault] TRACE 2022/03/22 19:33:20 {"data":{"attributes":{"action":"start"},"type":"ChargingStart"}}
--
{"type":"FUNCTIONAL","messages":[{"code":"err.func.403","message":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"errors":[{"errorCode":"err.func.403","errorMessage":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"error_reference":"FUNCTIONAL"}
[renault] TRACE 2022/03/22 19:33:20 GET https://accounts.eu1.gigya.com/accounts.login?apiKey=3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668&loginID=***&password=***
[renault] TRACE 2022/03/22 19:33:20 {
  "callId": "ae8d0718203f481488b5250a1c3cb012",
  "errorCode": 0,
  "apiVersion": 2,
  "statusCode": 200,
  "statusReason": "OK",
  "time": "2022-03-22T18:33:20.358Z",
  "registeredTimestamp": 1637926712,
  "UID": "60da6b4e-872f-4449-8614-6a6bc8f339cd",
  "UIDSignature": "eeyeYCGkHwvwhEFTFrx9dSw3sYk=",
  "signatureTimestamp": "1647974000",
  "created": "2021-11-26T11:38:32.292Z",
  "createdTimestamp": 1637926712,
  "isActive": true,
  "isRegistered": true,
  "isVerified": true,
  "lastLogin": "2022-03-22T18:33:20.324Z",
  "lastLoginTimestamp": 1647974000,
  "lastUpdated": "2021-11-26T11:38:47.701Z",
  "lastUpdatedTimestamp": 1637926727701,
  "loginProvider": "site",
  "oldestDataUpdated": "2021-11-26T11:38:32.292Z",
  "oldestDataUpdatedTimestamp": 1637926712292,
  "profile": {
    "email": "***"
  },
  "registered": "2021-11-26T11:38:32.390Z",
  "socialProviders": "site",
  "verified": "2021-11-26T11:38:47.701Z",
  "verifiedTimestamp": 1637926727701,
  "newUser": false,
  "sessionInfo": {
    "cookieName": "gac_3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668",
    "cookieValue": "st2.s.AcbHUh9wHg.RNQ-MNpXv0kfh9DexqIduKVDWm5uGxhOOhNRb1i5sm0dMVxgRCi1qza2M-Nb3kiqUiGh6qfGv35TA_ZYzUletMhNwBXX6NkivznbShOMmuo.SrEaawqez3259qEbAmu2pm_65pP2tDptymT33sBtM7SZ70hI8lUUn2DljQACgpy8XPyBCVGNo4iV98WDw70DkA.sc3"
  }
}
[renault] TRACE 2022/03/22 19:33:20 GET https://accounts.eu1.gigya.com/accounts.getJWT?apiKey=3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668&expiration=900&fields=data.personId%2Cdata.gigyaDataCenter&login_token=st2.s.AcbHUh9wHg.RNQ-MNpXv0kfh9DexqIduKVDWm5uGxhOOhNRb1i5sm0dMVxgRCi1qza2M-Nb3kiqUiGh6qfGv35TA_ZYzUletMhNwBXX6NkivznbShOMmuo.SrEaawqez3259qEbAmu2pm_65pP2tDptymT33sBtM7SZ70hI8lUUn2DljQACgpy8XPyBCVGNo4iV98WDw70DkA.sc3
[renault] TRACE 2022/03/22 19:33:20 {
  "callId": "ed1901aeda614639b55dacad80c073a0",
  "errorCode": 0,
  "apiVersion": 2,
  "statusCode": 200,
  "statusReason": "OK",
  "time": "2022-03-22T18:33:20.467Z",
  "ignoredFields": "",
  "id_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IlJFUTBNVVE1TjBOQ1JUSkVNemszTTBVMVJrTkRRMFUwUTBNMVJFRkJSamhETWpkRU5VRkJRZyJ9.eyJpc3MiOiJodHRwczovL2ZpZG0uZ2lneWEuY29tL2p3dC8zXzdQTGtzT3lCUmtIdjEyNng1V2hIYi01cHFDMXFGUjhwUWp4U2VMQjZuaEFuUEVSVFVsd25Zb3puSFN4d1g2NjgvIiwiYXBpS2V5IjoiM183UExrc095QlJrSHYxMjZ4NVdoSGItNXBxQzFxRlI4cFFqeFNlTEI2bmhBblBFUlRVbHduWW96bkhTeHdYNjY4IiwiaWF0IjoxNjQ3OTc0MDAwLCJleHAiOjE2NDc5NzQ5MDAsInN1YiI6IjYwZGE2YjRlLTg3MmYtNDQ0OS04NjE0LTZhNmJjOGYzMzljZCIsImRhdGEucGVyc29uSWQiOiJmOWVkOTE1MS1kNjEzLTRkZjYtOGIyOS1hZTdjNDQwNDFjNDIiLCJkYXRhLmdpZ3lhRGF0YUNlbnRlciI6ImV1MS5naWd5YS5jb20ifQ.i4NxSlZCl_cK0naUChkjh9t7ASwvnFr8CnUk2tgfhpgs-5YRr9F_8Hl-CgNEHUbl1hAY85nBPOC4QpW7LcIX3PlO4fvWLkLeKbM5mZjKPubMoTO4iENTnFo4ZrJAGbcR70Nw0zQV0euHjZB3ZZoDYnWYMhJYrCJpCqpOsOKuKoTfSfJY7vYihhdTeEIszImapiCcSz6A0s3j8Yd_tn1D9Tg2AsGtS7CCwKrIA_bj6hu7M-yRCe9MaeU0zaJg48WyQf18nnVgno7L3i7aUYVdknfTJ2wA6hdTQBLgaOmLt8dGJG6pcBc9PJejfthQvz89AoorOe5TycaKtJPxqCLK3A"
}
[renault] TRACE 2022/03/22 19:33:20 POST https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/00fcfbc7-633d-470e-84f3-98aac9b5e3ab/kamereon/kca/car-adapter/v1/cars/***/actions/charging-start?country=DE
[renault] TRACE 2022/03/22 19:33:20 {"data":{"attributes":{"action":"start"},"type":"ChargingStart"}}
--
{"type":"FUNCTIONAL","messages":[{"code":"err.func.403","message":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"errors":[{"errorCode":"err.func.403","errorMessage":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"error_reference":"FUNCTIONAL"}
[lp-1  ] ERROR 2022/03/22 19:33:20 wake-up vehicle: unexpected status: 403`

chrissooo commented:
Hier mal der complette Trace:

```
[renault] TRACE 2022/03/22 19:33:19 POST https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/00fcfbc7-633d-470e-84f3-98aac9b5e3ab/kamereon/kca/car-adapter/v1/cars/***/actions/charging-start?country=DE
[renault] TRACE 2022/03/22 19:33:20 {"data":{"attributes":{"action":"start"},"type":"ChargingStart"}}
--
{"type":"FUNCTIONAL","messages":[{"code":"err.func.403","message":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"errors":[{"errorCode":"err.func.403","errorMessage":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"error_reference":"FUNCTIONAL"}
[renault] TRACE 2022/03/22 19:33:20 GET https://accounts.eu1.gigya.com/accounts.login?apiKey=3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668&loginID=***&password=***
[renault] TRACE 2022/03/22 19:33:20 {
  "callId": "ae8d0718203f481488b5250a1c3cb012",
  "errorCode": 0,
  "apiVersion": 2,
  "statusCode": 200,
  "statusReason": "OK",
  "time": "2022-03-22T18:33:20.358Z",
  "registeredTimestamp": 1637926712,
  "UID": "60da6b4e-872f-4449-8614-6a6bc8f339cd",
  "UIDSignature": "eeyeYCGkHwvwhEFTFrx9dSw3sYk=",
  "signatureTimestamp": "1647974000",
  "created": "2021-11-26T11:38:32.292Z",
  "createdTimestamp": 1637926712,
  "isActive": true,
  "isRegistered": true,
  "isVerified": true,
  "lastLogin": "2022-03-22T18:33:20.324Z",
  "lastLoginTimestamp": 1647974000,
  "lastUpdated": "2021-11-26T11:38:47.701Z",
  "lastUpdatedTimestamp": 1637926727701,
  "loginProvider": "site",
  "oldestDataUpdated": "2021-11-26T11:38:32.292Z",
  "oldestDataUpdatedTimestamp": 1637926712292,
  "profile": {
    "email": "***"
  },
  "registered": "2021-11-26T11:38:32.390Z",
  "socialProviders": "site",
  "verified": "2021-11-26T11:38:47.701Z",
  "verifiedTimestamp": 1637926727701,
  "newUser": false,
  "sessionInfo": {
    "cookieName": "gac_3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668",
    "cookieValue": "st2.s.AcbHUh9wHg.RNQ-MNpXv0kfh9DexqIduKVDWm5uGxhOOhNRb1i5sm0dMVxgRCi1qza2M-Nb3kiqUiGh6qfGv35TA_ZYzUletMhNwBXX6NkivznbShOMmuo.SrEaawqez3259qEbAmu2pm_65pP2tDptymT33sBtM7SZ70hI8lUUn2DljQACgpy8XPyBCVGNo4iV98WDw70DkA.sc3"
  }
}
[renault] TRACE 2022/03/22 19:33:20 GET https://accounts.eu1.gigya.com/accounts.getJWT?apiKey=3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668&expiration=900&fields=data.personId%2Cdata.gigyaDataCenter&login_token=st2.s.AcbHUh9wHg.RNQ-MNpXv0kfh9DexqIduKVDWm5uGxhOOhNRb1i5sm0dMVxgRCi1qza2M-Nb3kiqUiGh6qfGv35TA_ZYzUletMhNwBXX6NkivznbShOMmuo.SrEaawqez3259qEbAmu2pm_65pP2tDptymT33sBtM7SZ70hI8lUUn2DljQACgpy8XPyBCVGNo4iV98WDw70DkA.sc3
[renault] TRACE 2022/03/22 19:33:20 {
  "callId": "ed1901aeda614639b55dacad80c073a0",
  "errorCode": 0,
  "apiVersion": 2,
  "statusCode": 200,
  "statusReason": "OK",
  "time": "2022-03-22T18:33:20.467Z",
  "ignoredFields": "",
  "id_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IlJFUTBNVVE1TjBOQ1JUSkVNemszTTBVMVJrTkRRMFUwUTBNMVJFRkJSamhETWpkRU5VRkJRZyJ9.eyJpc3MiOiJodHRwczovL2ZpZG0uZ2lneWEuY29tL2p3dC8zXzdQTGtzT3lCUmtIdjEyNng1V2hIYi01cHFDMXFGUjhwUWp4U2VMQjZuaEFuUEVSVFVsd25Zb3puSFN4d1g2NjgvIiwiYXBpS2V5IjoiM183UExrc095QlJrSHYxMjZ4NVdoSGItNXBxQzFxRlI4cFFqeFNlTEI2bmhBblBFUlRVbHduWW96bkhTeHdYNjY4IiwiaWF0IjoxNjQ3OTc0MDAwLCJleHAiOjE2NDc5NzQ5MDAsInN1YiI6IjYwZGE2YjRlLTg3MmYtNDQ0OS04NjE0LTZhNmJjOGYzMzljZCIsImRhdGEucGVyc29uSWQiOiJmOWVkOTE1MS1kNjEzLTRkZjYtOGIyOS1hZTdjNDQwNDFjNDIiLCJkYXRhLmdpZ3lhRGF0YUNlbnRlciI6ImV1MS5naWd5YS5jb20ifQ.i4NxSlZCl_cK0naUChkjh9t7ASwvnFr8CnUk2tgfhpgs-5YRr9F_8Hl-CgNEHUbl1hAY85nBPOC4QpW7LcIX3PlO4fvWLkLeKbM5mZjKPubMoTO4iENTnFo4ZrJAGbcR70Nw0zQV0euHjZB3ZZoDYnWYMhJYrCJpCqpOsOKuKoTfSfJY7vYihhdTeEIszImapiCcSz6A0s3j8Yd_tn1D9Tg2AsGtS7CCwKrIA_bj6hu7M-yRCe9MaeU0zaJg48WyQf18nnVgno7L3i7aUYVdknfTJ2wA6hdTQBLgaOmLt8dGJG6pcBc9PJejfthQvz89AoorOe5TycaKtJPxqCLK3A"
}
[renault] TRACE 2022/03/22 19:33:20 POST https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/00fcfbc7-633d-470e-84f3-98aac9b5e3ab/kamereon/kca/car-adapter/v1/cars/***/actions/charging-start?country=DE
[renault] TRACE 2022/03/22 19:33:20 {"data":{"attributes":{"action":"start"},"type":"ChargingStart"}}
--
{"type":"FUNCTIONAL","messages":[{"code":"err.func.403","message":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"errors":[{"errorCode":"err.func.403","errorMessage":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"error_reference":"FUNCTIONAL"}
[lp-1  ] ERROR 2022/03/22 19:33:20 wake-up vehicle: unexpected status: 403
```

chrissooo commented:
Reichen die Infos aus?

cow2k commented:
> Leider scheint es nicht richtig zu funktionieren. EInmal hat es geklappt. Dann kam aber auch folgender fehler kurz zuvor: [lp-1 ] ERROR 2022/03/22 15:00:10 wake-up vehicle: Post "https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/XXXXXXXXXXXXXXX/kamereon/kca/car-adapter/v1/cars/XXXXXXXXXXX/actions/charging-start?country=DE": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
> 
> Jetzt kommt nur noch [lp-1 ] ERROR 2022/03/22 16:05:26 wake-up vehicle: unexpected status: 403

Scheinbar  wurde der API Key von Renault geändert. Der neue ist wohl: apikey: VAX7XYKGfa92yMvXculCkEFyfZbuM7Ss

vielleicht kannst du es ja damit nochmal probieren?

cow2k commented on vehicle/renault.go:
```diff
@@ -270,15 +276,15 @@ func (v *Renault) jwtToken(sessionCookie string) (string, error) {
+	params := url.Values{"country": []string{"DE"}}
 	headers := map[string]string{
 		"x-gigya-id_token": v.gigyaJwtToken,
 		"apikey":           "Ae9FDWugRxZQAGm3Sxgk7uJn6Q4CGEA2", // v.kamereon.APIKey
```
Api Key changed. New one seems to be VAX7XYKGfa92yMvXculCkEFyfZbuM7Ss (see https://github.com/jamesremuscat/pyze/issues/109#issuecomment-1073218876)

andig commented:
@premultiply was fehlt hier noch?

chrissooo commented:
Ich könnte noch die Änderungen von cow2k testen ... versuche das noch vor dem Wochenende zu schaffen.

@cow2k der Key war zu dem Zeitpunkt schon der richtige. Er konnte im gleichen Test zumindest die Daten des Fahrzeugs abrufen was ja über den gleichen Weg funktioniert. Aber du hast jetzt noch andere Dinge geändert, oder?

premultiply commented:
@andig Mir fehlt hier eigentlich nur dass jemand bestätigt dass es funktioniert oder sagt was genau nicht funktioniert und ggf. verbessert werden kann.

chrissooo commented:
@premultiply Ich hatte ja schon deine Version oben getestet. Das hat ja nicht funktioniert. Logfile habe ich ja bereitgestellt. Jetzt hat @cow2k ja nochmal paar Dinge geändert. Das würde ich jetzt so schnell wie es geht machen und berichten.

chrissooo commented:
@cow2k das mit dem API Key ist nicht schlimm. Ich habe das ja in das Release 0.90 gemerged. Der neue API Key ist dann richtig. Habe das mal überprüft.

chrissooo commented:
Habe es doch schnell jetzt getestet. Also wie gesagt es funktioniert nicht.

Es erscheint die Meldung: [lp-1  ] ERROR 2022/04/21 10:38:03 wake-up vehicle: unexpected status: 403

Die Anmeldunge bei Rennault geht aber grundsätzlich. Weil die Daten (Akkustand usw.) beim Programstart ausgelesen werden. Es muss an etwas anderem liegen.

Betroffener Auszug vom Logfile:
`[site  ] DEBUG 2022/04/21 10:37:54 ----
[heidel] TRACE 2022/04/21 10:37:54 modbus: send 01 04 00 0e 00 01 50 09
[heidel] TRACE 2022/04/21 10:37:55 modbus: recv 01 04 02 00 00 b9 30
[lp-1  ] DEBUG 2022/04/21 10:37:55 charge power: 0W
[cache ] TRACE 2022/04/21 10:37:55 lp-1/chargePower: 0
[http  ] TRACE 2022/04/21 10:37:55 GET http://192.168.178.134/api/dxs.json?dxsEntries=67109120
[http  ] TRACE 2022/04/21 10:37:55 {"dxsEntries":[{"dxsId":67109120,"value":7915.906738}],"session":{"sessionId":0,"roleId":0},"status":{"code":0}}
[site  ] DEBUG 2022/04/21 10:37:55 pv power: 7916W
[cache ] TRACE 2022/04/21 10:37:55 pvPower: 7915.906738
[modbus] TRACE 2022/04/21 10:37:55 modbus: send 00 26 00 00 00 06 47 03 9c 87 00 69
[modbus] TRACE 2022/04/21 10:37:55 modbus: recv 00 26 00 00 00 d5 47 03 d2 80 00 04 48 04 3f 04 49 ff fe 80 00 5b be 5c 04 5b d6 80 00 80 00 80 00 80 00 ff fe 13 85 ff fe fd 01 ff 00 ff 02 fe ff 00 01 fd 00 ff 00 ff 01 fe ff 00 01 ff e4 ff f9 ff ed ff ff 00 01 03 e7 03 e8 03 e5 03 e8 ff fd ff ce 4d e6 ff f0 8f ab ff ed fb b6 ff ed 79 e6 00 09 93 17 00 06 7a 29 00 02 87 c5 00 02 d9 c9 00 00 ff cd cb 8f ff f0 65 64 ff ed c6 a7 ff ed 64 f2 00 0e d2 79 00 08 44 58 00 05 48 02 00 04 0c 77 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 00 00 00 00 00 00
[site  ] DEBUG 2022/04/21 10:37:55 grid power: -7670W
[site  ] DEBUG 2022/04/21 10:37:55 site power: -7670W
[heidel] TRACE 2022/04/21 10:37:55 modbus: send 01 04 00 06 00 01 d1 cb
[cache ] TRACE 2022/04/21 10:37:55 gridPower: -7670
[cache ] TRACE 2022/04/21 10:37:55 lp-1/mode: pv
[heidel] TRACE 2022/04/21 10:37:55 modbus: recv 01 04 02 00 00 b9 30
[heidel] TRACE 2022/04/21 10:37:55 modbus: send 01 04 00 07 00 01 80 0b
[heidel] TRACE 2022/04/21 10:37:55 modbus: recv 01 04 02 00 00 b9 30
[heidel] TRACE 2022/04/21 10:37:55 modbus: send 01 04 00 08 00 01 b0 08
[heidel] TRACE 2022/04/21 10:37:55 modbus: recv 01 04 02 00 00 b9 30
[lp-1  ] DEBUG 2022/04/21 10:37:55 charge currents: [0 0 0]A
[cache ] TRACE 2022/04/21 10:37:55 lp-1/chargeCurrents: [0 0 0]
[cache ] TRACE 2022/04/21 10:37:55 lp-1/chargeCurrent: 16
[cache ] TRACE 2022/04/21 10:37:55 lp-1/chargedEnergy: 7.999999999924512
[heidel] TRACE 2022/04/21 10:37:55 modbus: send 01 04 00 05 00 01 21 cb
[cache ] TRACE 2022/04/21 10:37:55 lp-1/chargeDuration: 14s
[heidel] TRACE 2022/04/21 10:37:55 modbus: recv 01 04 02 00 05 79 33
[lp-1  ] DEBUG 2022/04/21 10:37:55 charger status: B
[lp-1  ] DEBUG 2022/04/21 10:37:55 next soc poll remaining time: 58m15s
[heidel] TRACE 2022/04/21 10:37:55 modbus: send 01 03 01 05 00 01 95 f7
[cache ] TRACE 2022/04/21 10:37:55 lp-1/connected: true
[cache ] TRACE 2022/04/21 10:37:55 lp-1/charging: false
[cache ] TRACE 2022/04/21 10:37:55 lp-1/enabled: true
[heidel] TRACE 2022/04/21 10:37:55 modbus: recv 01 03 02 00 a0 b8 3c
[lp-1  ] DEBUG 2022/04/21 10:37:55 pv charge current: 33.3A = 0A + 33.3A (-7670W @ 1p)
[lp-1  ] DEBUG 2022/04/21 10:37:55 climater active: false, target temp: 20.0°C, outside temp: 0.0°C
[cache ] TRACE 2022/04/21 10:37:55 homePower: 245.90673799999968
[cache ] TRACE 2022/04/21 10:37:55 lp-1/climater: off
[modbus] TRACE 2022/04/21 10:38:00 modbus: closing connection due to idle timeout: 5.000180308s
[site  ] DEBUG 2022/04/21 10:38:01 ----
[heidel] TRACE 2022/04/21 10:38:01 modbus: send 01 04 00 0e 00 01 50 09
[heidel] TRACE 2022/04/21 10:38:02 modbus: recv 01 04 02 00 00 b9 30
[lp-1  ] DEBUG 2022/04/21 10:38:02 charge power: 0W
[http  ] TRACE 2022/04/21 10:38:02 GET http://192.168.178.134/api/dxs.json?dxsEntries=67109120
[cache ] TRACE 2022/04/21 10:38:02 lp-1/chargePower: 0
[http  ] TRACE 2022/04/21 10:38:02 {"dxsEntries":[{"dxsId":67109120,"value":7919.091797}],"session":{"sessionId":0,"roleId":0},"status":{"code":0}}
[site  ] DEBUG 2022/04/21 10:38:02 pv power: 7919W
[cache ] TRACE 2022/04/21 10:38:02 pvPower: 7919.091797
[modbus] TRACE 2022/04/21 10:38:02 modbus: send 00 27 00 00 00 06 47 03 9c 87 00 69
[modbus] TRACE 2022/04/21 10:38:02 modbus: recv 00 27 00 00 00 d5 47 03 d2 80 00 04 4a 04 44 04 49 ff fe 80 00 5b cc 5c 0a 5b e0 80 00 80 00 80 00 80 00 ff fe 13 85 ff fe fc ff ff 00 ff 01 fe ff 00 01 fc fe ff 00 ff 00 fe ff 00 01 ff ec ff fb ff f0 00 01 00 01 03 e8 03 e8 03 e6 03 e8 ff fd ff ce 4d d7 ff f0 8f a5 ff ed fb b1 ff ed 79 e1 00 09 93 17 00 06 7a 29 00 02 87 c5 00 02 d9 c9 00 00 ff cd cb 80 ff f0 65 5f ff ed c6 a2 ff ed 64 ec 00 0e d2 79 00 08 44 58 00 05 48 02 00 04 0c 77 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 80 00 00 00 00 00 00 00 00 00
[site  ] DEBUG 2022/04/21 10:38:02 grid power: -7690W
[site  ] DEBUG 2022/04/21 10:38:02 site power: -7690W
[cache ] TRACE 2022/04/21 10:38:02 gridPower: -7690
[heidel] TRACE 2022/04/21 10:38:02 modbus: send 01 04 00 06 00 01 d1 cb
[cache ] TRACE 2022/04/21 10:38:02 lp-1/mode: pv
[heidel] TRACE 2022/04/21 10:38:02 modbus: recv 01 04 02 00 00 b9 30
[heidel] TRACE 2022/04/21 10:38:02 modbus: send 01 04 00 07 00 01 80 0b
[heidel] TRACE 2022/04/21 10:38:02 modbus: recv 01 04 02 00 00 b9 30
[heidel] TRACE 2022/04/21 10:38:02 modbus: send 01 04 00 08 00 01 b0 08
[heidel] TRACE 2022/04/21 10:38:02 modbus: recv 01 04 02 00 00 b9 30
[lp-1  ] DEBUG 2022/04/21 10:38:02 charge currents: [0 0 0]A
[cache ] TRACE 2022/04/21 10:38:02 lp-1/chargeCurrents: [0 0 0]
[heidel] TRACE 2022/04/21 10:38:02 modbus: send 01 04 00 05 00 01 21 cb
[cache ] TRACE 2022/04/21 10:38:02 lp-1/chargeCurrent: 16
[cache ] TRACE 2022/04/21 10:38:02 lp-1/chargedEnergy: 7.999999999924512
[cache ] TRACE 2022/04/21 10:38:02 lp-1/chargeDuration: 14s
[heidel] TRACE 2022/04/21 10:38:02 modbus: recv 01 04 02 00 05 79 33
[lp-1  ] DEBUG 2022/04/21 10:38:02 charger status: B
[cache ] TRACE 2022/04/21 10:38:02 lp-1/connected: true
[cache ] TRACE 2022/04/21 10:38:02 lp-1/charging: false
[lp-1  ] DEBUG 2022/04/21 10:38:02 next soc poll remaining time: 58m8s
[heidel] TRACE 2022/04/21 10:38:02 modbus: send 01 03 01 05 00 01 95 f7
[cache ] TRACE 2022/04/21 10:38:02 lp-1/enabled: true
[heidel] TRACE 2022/04/21 10:38:02 modbus: recv 01 03 02 00 a0 b8 3c
[lp-1  ] DEBUG 2022/04/21 10:38:02 pv charge current: 33.4A = 0A + 33.4A (-7690W @ 1p)
[lp-1  ] DEBUG 2022/04/21 10:38:02 climater active: false, target temp: 20.0°C, outside temp: 0.0°C
[cache ] TRACE 2022/04/21 10:38:02 lp-1/climater: off
[renault] TRACE 2022/04/21 10:38:02 POST https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/00fcfbc7-633d-470e-84f3-98aac9b5e3ab/kamereon/kca/car-adapter/v1/cars/***/actions/charging-start?country=DE
[renault] TRACE 2022/04/21 10:38:02 {"data":{"attributes":{"action":"start"},"type":"ChargingStart"}}
--
{"type":"FUNCTIONAL","messages":[{"code":"err.func.403","message":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"errors":[{"errorCode":"err.func.403","errorMessage":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"error_reference":"FUNCTIONAL"}
[renault] TRACE 2022/04/21 10:38:02 GET https://accounts.eu1.gigya.com/accounts.login?apiKey=3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668&loginID=***&password=***
[renault] TRACE 2022/04/21 10:38:02 {
  "callId": "20825df3ea8b4961ba1cbb8a63875c67",
  "errorCode": 0,
  "apiVersion": 2,
  "statusCode": 200,
  "statusReason": "OK",
  "time": "2022-04-21T08:38:02.766Z",
  "registeredTimestamp": 1637926712,
  "UID": "60da6b4e-872f-4449-8614-6a6bc8f339cd",
  "UIDSignature": "k5zVZtoJvKgE9J31iFCGlVJeOOQ=",
  "signatureTimestamp": "1650530282",
  "created": "2021-11-26T11:38:32.292Z",
  "createdTimestamp": 1637926712,
  "isActive": true,
  "isRegistered": true,
  "isVerified": true,
  "lastLogin": "2022-04-21T08:38:02.730Z",
  "lastLoginTimestamp": 1650530282,
  "lastUpdated": "2021-11-26T11:38:47.701Z",
  "lastUpdatedTimestamp": 1637926727701,
  "loginProvider": "site",
  "oldestDataUpdated": "2021-11-26T11:38:32.292Z",
  "oldestDataUpdatedTimestamp": 1637926712292,
  "profile": {
    "email": "***"
  },
  "registered": "2021-11-26T11:38:32.390Z",
  "socialProviders": "site",
  "verified": "2021-11-26T11:38:47.701Z",
  "verifiedTimestamp": 1637926727701,
  "newUser": false,
  "sessionInfo": {
    "cookieName": "gac_3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668",
    "cookieValue": "st2.s.AcbHeJ77Fg.lXwEo56DaSWqOSw-25rZQYhCAXbWT7WfcW5854sCVlk-BZehHg9j4ETyrQ7W9gjQhy6mVxdKjkIl-opOZuXUePirHqxkXGkB6zyrLM7k6Q8.Mp1g77I9YiNq0HeSjkxpoB4Cjm9cjIfyJWW8kkVk_Bd0pQx8ldUQOJ8a1m38M5KB3-CKlDiD6usjTssVZs2RZw.sc3"
  }
}
[renault] TRACE 2022/04/21 10:38:02 GET https://accounts.eu1.gigya.com/accounts.getJWT?apiKey=3_7PLksOyBRkHv126x5WhHb-5pqC1qFR8pQjxSeLB6nhAnPERTUlwnYoznHSxwX668&expiration=900&fields=data.personId%2Cdata.gigyaDataCenter&login_token=st2.s.AcbHeJ77Fg.lXwEo56DaSWqOSw-25rZQYhCAXbWT7WfcW5854sCVlk-BZehHg9j4ETyrQ7W9gjQhy6mVxdKjkIl-opOZuXUePirHqxkXGkB6zyrLM7k6Q8.Mp1g77I9YiNq0HeSjkxpoB4Cjm9cjIfyJWW8kkVk_Bd0pQx8ldUQOJ8a1m38M5KB3-CKlDiD6usjTssVZs2RZw.sc3
[renault] TRACE 2022/04/21 10:38:02 {
  "callId": "43be3f08bff94ec18fd9a8433dc48849",
  "errorCode": 0,
  "apiVersion": 2,
  "statusCode": 200,
  "statusReason": "OK",
  "time": "2022-04-21T08:38:02.863Z",
  "ignoredFields": "",
  "id_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IlJFUTBNVVE1TjBOQ1JUSkVNemszTTBVMVJrTkRRMFUwUTBNMVJFRkJSamhETWpkRU5VRkJRZyJ9.eyJpc3MiOiJodHRwczovL2ZpZG0uZ2lneWEuY29tL2p3dC8zXzdQTGtzT3lCUmtIdjEyNng1V2hIYi01cHFDMXFGUjhwUWp4U2VMQjZuaEFuUEVSVFVsd25Zb3puSFN4d1g2NjgvIiwiYXBpS2V5IjoiM183UExrc095QlJrSHYxMjZ4NVdoSGItNXBxQzFxRlI4cFFqeFNlTEI2bmhBblBFUlRVbHduWW96bkhTeHdYNjY4IiwiaWF0IjoxNjUwNTMwMjgyLCJleHAiOjE2NTA1MzExODIsInN1YiI6IjYwZGE2YjRlLTg3MmYtNDQ0OS04NjE0LTZhNmJjOGYzMzljZCIsImRhdGEucGVyc29uSWQiOiJmOWVkOTE1MS1kNjEzLTRkZjYtOGIyOS1hZTdjNDQwNDFjNDIiLCJkYXRhLmdpZ3lhRGF0YUNlbnRlciI6ImV1MS5naWd5YS5jb20ifQ.pCA2KLfDPo_HaA6MQhaZpy_ppp_oyd4mrlc86QKAIOhx6dO_Ld4b5YYu---3LXy7AuCCu1opMlfQUTnq_SJ4alQXs3yoODxM765iUuH731rasv0q9rtTESWXvvW-vOlHdI3xUd3XY7TJ_Ptf9hJlHJnusLhjRfMR4mrNsAqjm-Q0PNxKjCU2pSL-ftMsfd-0GNDxkWqXnZ60WCwhHOjz3nOT0cOTUFIzfwyPEIu1K7QMfEIVKh2SoBkfnX_kYZiO9kPoLidvtNS6UB_NwKAHI_o95VfgP8zMX0BJs6jwhBFLHNE6oqoP6DVLe9OQ62EKvg9xMGBBDXvFQuoOj1rGHQ"
}
[renault] TRACE 2022/04/21 10:38:02 POST https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/00fcfbc7-633d-470e-84f3-98aac9b5e3ab/kamereon/kca/car-adapter/v1/cars/***/actions/charging-start?country=DE
[renault] TRACE 2022/04/21 10:38:03 {"data":{"attributes":{"action":"start"},"type":"ChargingStart"}}
--
{"type":"FUNCTIONAL","messages":[{"code":"err.func.403","message":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"errors":[{"errorCode":"err.func.403","errorMessage":"{\"errors\":[{\"status\":\"403\",\"code\":\"security.access\",\"detail\":\"Access is denied for this resource\"}]}"}],"error_reference":"FUNCTIONAL"}
[lp-1  ] ERROR 2022/04/21 10:38:03 wake-up vehicle: unexpected status: 403
[cache ] TRACE 2022/04/21 10:38:03 homePower: 229.09179700000004`

andig commented:
@chrissooo keine Tokens posten!

@premultiply der Test kann natürlich failen wenn man den PR auscheckt und da der falsche Key drin ist. Rebase?

chrissooo commented:
> @chrissooo keine Tokens posten!

Ok, ich dachte dass im logfile Zugangsdaten usw. immer rausgenommen werden.
 
> @premultiply der Test kann natürlich failen wenn man den PR auscheckt und da der falsche Key drin ist. Rebase?

Wie gesagt habe ich darauf geachtet. Bei meinem Test ist der richtige Key drin, trotzdem ist das könnte man das natürlich korrigieren.

chrissooo commented:
[renault-wakeup-test.log](https://github.com/evcc-io/evcc/files/8530493/renault-wakeup-test.log)

Habe hier mal ein komplettes Logfile erstellt. Vom Start von evcc an. Man sieht dass die API grundsätzlich funktioniert, also die Daten des Fahrzeugs abgerufen werden können. Der StartWakeup schlägt dann mit 403 Fehler fehl.

cow2k commented:
> Ich könnte noch die Änderungen von cow2k testen ... versuche das noch vor dem Wochenende zu schaffen.
> 
> @cow2k der Key war zu dem Zeitpunkt schon der richtige. Er konnte im gleichen Test zumindest die Daten des Fahrzeugs abrufen was ja über den gleichen Weg funktioniert. Aber du hast jetzt noch andere Dinge geändert, oder?

Ne, ich habe gar nichts geändert. Dazu reichen meine Git skills nicht aus :-( Mir ist nur beim drüber Gucken über die Changes aufgefallen, dass der Key noch der Alte war. Leider habe ich damit wohl nicht den Kern des Problems erkannt. Ich taste mich jetzt gerade an die Renault API ran und probiere erstmal mittels https://github.com/hacf-fr/renault-api ob ich bei meiner Zoe damit den Ladevorgang starten kann. Das ist zumindest mal in Python und damit eher meine Welt :-) Wenn das klappt, wage ich mich mal an Go und probiere den vorliegenden PR mit evcc aus.

chrissooo commented:
@premultiply kann ich sonst noch was tun? Langt dir das komplette Logfiles? Brauchst du sonst noch was?

premultiply commented:
Der Ansatz von @cow2k ist ganz gut denke ich. Mit den diversen sonstigen Tools/Apps mal versuchen das Fahrzeug zu wecken und dies dann bei Erfolg genau anschauen wie es dort implementiert ist.

chrissooo commented:
@premultiply @cow2k ich habs ... wusste nicht dass man bei hacf-fr ein commandline Tool für die Api installieren kann. Damit konnte ich das Fahrzeug tatsächlich wecken. Es liegt daran dass hier die action: start verwendet wird und mit einer anderen URL.

Benötigt wird hingegen action: resume, type: ChargePauseResume, URL: /commerce/v1/accounts/%s/kamereon/kcm/v1/vehicles/%s/charge/pause-resume

Damit funktioniert es auch im evcc! Habe die Änderungen lokal durchgeführt und getestet.

chrissooo commented:
Was nun? Ich habe die Änderungen mal in den PR commited. Kann die jemand hier mergen? Dann würde ich das hier final testen.

premultiply commented:
Ich seh nix. Du müsstest bitte einen PR gegen den Branch hier machen.

chrissooo commented:
Ok, habe ich das nun richtig gemacht? https://github.com/evcc-io/evcc/pull/3218

premultiply commented:
Ja, genau. Da sind aber noch Fehler bei den Checks aufgetreten.

chrissooo committed:
Update WakeUp call (#3218)

To wakeup a car (renault api) we must use "resume" with tyle "ChargePauseResume" an a another uri.

chrissooo commented:
Hallo,
der Test hat funktioniert. Dazu habe ich bei der Heidelberg Box die WakeUp Funktion auskommentiert damit ich das testen kann. Der WakeUp an der Box wird ja als erstes ausgeführt, richtig?

Den echten Test, also dass der WakeUp an der Box fehlschlägt und er dann den Fahrzeug WakeUp mach, kann ich erst am Wochenende testen. Das Auto muss nämlich paar Stunden an der Wallbox hängen um in den Tiefschlaf zu gelangen damit der CP Interrupt nicht mehr funktioniert und er dann über RenaultAPI gehen sollte.

Könnte man eventuell die Möglichkeit bieten per Konfigurationsdatei folgendes einzustellen?
- WakeUp Wallbox und WakeUp Auto per Konfiguration einzeln an- und abschaltbar
- Reihenfolge per Konfiguration beeinflussbar, also welcher WakeUp zuerst gemacht werden soll? (Renault WakeUp ist schneller :-))

chrissooo commented:
@andig ich habe mir mal den Code ein wenig angeschaut. Verstehe ich das richtig?

In der loadpint.go gibt es ja die methode die das ganze antriggert:
```
func (lp *LoadPoint) wakeUpVehicle() {
	// charger
	if c, ok := lp.charger.(api.AlarmClock); ok {
		if err := c.WakeUp(); err != nil {
			lp.log.ERROR.Printf("wake-up charger: %v", err)
		}
		return
	}

	// vehicle
	if lp.vehicle != nil {
		if vs, ok := lp.vehicle.(api.AlarmClock); ok {
			if err := vs.WakeUp(); err != nil {
				lp.log.ERROR.Printf("wake-up vehicle: %v", err)
			}
		}
	}
}
```

Aufgerufen wird es durch:
```
	// Wake-up checks
	if lp.enabled && lp.status == api.StatusB &&
		int(lp.vehicleSoc) < lp.SoC.Target && lp.wakeUpTimer.Expired() {
		lp.wakeUpVehicle()
	}
```

Wenn ich jetzt so einen Fall habe wie bei mir, dass unter bestimmten Umständen (tiefer Schlaf des Autos nach mehreren Stunden an der Box ohne Laden) das Auto über die Wallbox nicht aufgeweckt werden kann mit dem charger WakeUp call, dass der vehicle WakeUp als zweiter WakeUp gar nicht aufgerufen wird sondern nur dann wenn es garkein Charger WakeUp gibt?

andig commented:
Ja, das müssen wir noch ändern

chrissooo commented:
> Ja, das müssen wir noch ändern

Super. Und wie gesagt ... dieser PR, also Renault WakeUP funktioniert.

chrissooo commented:
@andig danke fürs mergen.

Wegen dem Thema wenn der LP WakeUp nicht geht soll dann der Vehicel WakeUp gezündet werden. Kann ich das irgendwie vorantreiben? Selber coden wird glaube ich nix.

DaEffendi commented:
Hallo, 
funktioniert das jetzt mit dem API wakeup für Renault. Ich finde dazu keine Anleitung. 
Dankeschön

chrissooo commented:
Ja, funktioniert.

DaEffendi commented:
> Ja, funktioniert.

Kannst du mir sagen wo ich Doku finde oder was ich einstellen muss?
